src/components/Books.js

```diff
@@ -1,13 +1,38 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import store from '../redux/configureStore';
 import CreateNewBook from './CreateNewBook';
 import BooksList from './BooksList';
+import { getList } from '../redux/books/books';
 
 const Books = () => {
+  const dispatch = useDispatch();
+
   const [books, setBooks] = useState([]);
 
   store.subscribe(() => setBooks(store.getState().booksReducer));
 
+  const getBookList = async () => {
+    await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/eTfxKsR9wf3ubgm9W4xT/books')
+      .then((response) => response.json())
+      .then((json) => {
+        const newArray = [];
+        Object.entries(json).forEach((book) => {
+          const newBook = {
+            item_id: book[0],
+            title: book[1][0].title,
+            category: book[1][0].category,
+          };
+          newArray.push(newBook);
+        });
+        dispatch(getList(newArray));
+      });
+  };
+
+  useEffect(() => {
+    getBookList();
+  }, []);
```

Comment:
**Optional Changes**
- [ ] The `console` throws a warning ` Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup`. You can checkout this [`stackoverflow` answer ](https://stackoverflow.com/a/60907638/5060694)to see how to go about fixing it.